### PR TITLE
Myeloid update of coverage2json script

### PR DESCRIPTION
**Pull request by kgrimes8**
## Change description
- Changed `coverage2json.py` script so that the hotspots and/or genescreen bed files are optional

## Justification for minor update
Doesn’t affect sensitivity – changes are only formatting of database coverage input.

## Testing description
1. code review should involve running the script with:
- combined, gene screen and hotspots present
- combined and gene screen present
- combined and hotspots present
- only combined present
2. check that coverage files from a previous run are identical using the new script

## Test data location
All test data for this verification is saved on Wren: `/Output/validations/TSO500/TSO500_coverage_script_update/`

This run was used to test the script:
- 220406_A01771_0010_BH5VCMDRX2

## Verification results
1. ran as expected, see `Database_missing_genescreen_or_hotspot` in the test data
2. ran as expected, `Database_new` and `Database_old` in the test data are replicates of a whole run with the old and new version of the script, all files were identical as verified by md5sum

### CODE REVIEW

File: scripts/coverage2json.py [hunk 1]
```diff
@@ -262,36 +326,64 @@ def parse_sample_data(sample_coverage_folder, referral_type):
...
-	sample_region_df = pd.merge(sample_270_region_df, sample_135_region_df, how = 'outer', on = ['CHR', 'START', 'END', 'META', 'AVG_DEPTH'])
-	sample_gene_df = pd.merge(sample_270_gene_df, sample_135_gene_df, how = 'outer', on = ['GENE', 'AVG_DEPTH'])
+	## join hotspots if present, remove nan values and round to integer, else make empty df
+	if hotspots_present:
+		sample_hotspot_region_df = pd.merge(sample_270_hotspot_region_df, sample_135_hotspot_region_df, how = 'outer', on = ['CHR', 'START', 'END', 'META', 'AVG_DEPTH'])
+		sample_hotspot_region_df['AVG_DEPTH'] = sample_hotspot_region_df['AVG_DEPTH'].apply(lambda x: None if np.isnan(x) else int((Decimal(str(x)).quantize(Decimal('1')))))
+		sample_hotspot_region_df['PERC_COVERAGE@270'] = sample_hotspot_region_df['PERC_COVERAGE@270'].apply(lambda x: None if np.isnan(x) else int((Decimal(str(x)).quantize(Decimal('1')))))
+		sample_hotspot_region_df['PERC_COVERAGE@135'] = sample_hotspot_region_df['PERC_COVERAGE@135'].apply(lambda x: None if np.isnan(x) else int((Decimal(str(x)).quantize(Decimal('1')))))
+	else:
+		sample_hotspot_region_df = pd.Dataframe()
```
Comment (erikwaskiewicz) [inline scripts/coverage2json.py:357]: Should be capital F

File: scripts/coverage2json.py [hunk 2]
```diff
@@ -262,36 +326,64 @@ def parse_sample_data(sample_coverage_folder, referral_type):
...
 
+
+	## join genescreen if present, remove nan values and round to integer, else make empty df
+	if genescreen_present:
+		sample_genescreen_region_df = pd.merge(sample_270_genescreen_region_df, sample_135_genescreen_region_df, how = 'outer', on = ['CHR', 'START', 'END', 'META', 'AVG_DEPTH'])
+		sample_genescreen_region_df['AVG_DEPTH'] = sample_genescreen_region_df['AVG_DEPTH'].apply(lambda x: None if np.isnan(x) else int((Decimal(str(x)).quantize(Decimal('1')))))
+		sample_genescreen_region_df['PERC_COVERAGE@270'] = sample_genescreen_region_df['PERC_COVERAGE@270'].apply(lambda x: None if np.isnan(x) else int((Decimal(str(x)).quantize(Decimal('1')))))
+		sample_genescreen_region_df['PERC_COVERAGE@135'] = sample_genescreen_region_df['PERC_COVERAGE@135'].apply(lambda x: None if np.isnan(x) else int((Decimal(str(x)).quantize(Decimal('1')))))
+	else:
+		sample_genescreen_region_df = pd.Dataframe()
```
Comment (erikwaskiewicz) [inline scripts/coverage2json.py:367]: Should be capital F

File: scripts/coverage2json.py [hunk 3]
```diff
@@ -322,13 +414,17 @@ def create_output_dict(gene_list, main_gene_df, genescreen_region_df, hotspots_r
...
+				print(item)
+				# check if gene is the gene in the 4th column then add
+				if key == item[3].split('(')[0]:
+					filtered_hotspot_region_list.append(item)
+			output_dict[key]['hotspot_regions'] = filtered_hotspot_region_list
+		else:
+			output_dict[key]['hotspot_regions'] = []
 
 		## check if any gaps and add if there are
 		if len(sample_135_gaps_df) > 0:
```
Comment (erikwaskiewicz) [inline scripts/coverage2json.py:430]: Gaps section should be indented so only filled out when there is a hotspots coverage file

**Review by erikwaskiewicz — APPROVED** [review]
Some comments but otherwise all good. Kal's left the service so I'll make changes and ask for a review of specific changes in the comments

**Comment by erikwaskiewicz** [comment]
@SophieS9 can you review the changes from commit [d7426ec](https://github.com/AWGL/TSO500_post_processing/pull/24/commits/d7426eced0c8414e95a52f51fe28a3b883ded4b9) only please?

**Review by SophieS9 — APPROVED** [review]
All good - small syntax changes and moving an if statement to be within the correct loop

**Pull request merged**